how_to_solve_a_problem.md

```diff
@@ -44,6 +44,24 @@ In any case, as always, be sure to reach out to other team members to see if the
 
 To emphasize, a workaround is a valid solution, but it's kept as a last resort. Having a workaround in place often adds a technical debt to the system. Meaning, you have solved a problem, but most likely created new future ones, so be careful with this solution.
 
+### Taking breaks
+
+With the exception of an urgent bug on a production site, sometimes a complicated problem needs to be set aside. That is, as long as you are within the timeboxed range, you may decide to let things go until the next day.
+
+Taking a break for a drink or even a good night sleep can be  more valuable than sitting for hours in front of the computer fighting it. Find the balance between not giving up and temporarily letting go.
+
+More often than not we are able to solve a problem by not really thinking about it.
+
+### Multiple bugs
+
+The hardest problems to solve are the ones that are composed of multiple bugs. You can never know upfront if a bug is actually multiple bugs. Be disciplined by not jumping to conclusions and always *verifing your assumptions*.
+
+For example, if you expect to see some value on the screen, don't jump to the conclusion that there is a problem with the view related code. Are you sure that the value you expect to see even exists? Verify it by starting at the root. Maybe the problem is in the data - it wasn't saved correctly. Go ahead and fix that.
```

Comment:
The intent is root of the program (starting from the beginning - back-end
to front-end), not the root of the bug.  Why should they start at the data
before the front-end?  I lack tech-speak. ;)



Rachel Baram

Marketing and Business Development
Gizra Internet Solutions Ltd.
www.gizra.com cell:+972-54-2270555


On Wed, Jun 22, 2016 at 1:26 PM, Amitai Burstein <notifications@github.com>
wrote:

> In how_to_solve_a_problem.md
> <https://github.com/Gizra/Gizra-Way-Book/pull/28#discussion_r68029682>:
>
> > @@ -44,6 +44,24 @@ In any case, as always, be sure to reach out to other team members to see if the
> >
> >  To emphasize, a workaround is a valid solution, but it's kept as a last resort. Having a workaround in place often adds a technical debt to the system. Meaning, you have solved a problem, but most likely created new future ones, so be careful with this solution.
> >
> > +### Taking breaks
> > +
> > +With the exception of an urgent bug on a production site, sometimes a complicated problem needs to be set aside. That is, as long as you are within the timeboxed range, you may decide to let things go until the next day.
> > +
> > +Taking a break for a drink or even a good night sleep can be  more valuable than sitting for hours in front of the computer fighting it. Find the balance between not giving up and temporarily letting go.
> > +
> > +More often than not we are able to solve a problem by not really thinking about it.
> > +
> > +### Multiple bugs
> > +
> > +The hardest problems to solve are the ones that are composed of multiple bugs. You can never know upfront if a bug is actually multiple bugs. Be disciplined by not jumping to conclusions and always *verifing your assumptions*.
> > +
> > +For example, if you expect to see some value on the screen, don't jump to the conclusion that there is a problem with the view related code. Are you sure that the value you expect to see even exists? Verify it by starting at the root. Maybe the problem is in the data - it wasn't saved correctly. Go ahead and fix that.
>
> But you don't know if it's the root ;)
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/Gizra/Gizra-Way-Book/pull/28/files/d0c2af4009408c5da3594f48ab1113c5de6dcd45#r68029682>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/ANpkXydI4fncjd6UBg57UZMGbLK5JBgqks5qOQ3mgaJpZM4I7gcV>
> .
>